tests/ClientTest.php

```diff
@@ -68,13 +70,16 @@ public function testConstructClient()
         $client = new MockClient('', $handler);
         $client->runRawQuery('query_string',  false, ['name' => 'val']);
 
+        $client = new MockClient('', $handler, [], ['headers' => [ 'Authorization' => 'Basic xyz', 'User-Agent' => 'test' ]]);
```

Comment:
@actionm one small thing, we need to test that the httpOptions argument overrixes the values in the authorizationHeaders argument. Can you please provide an array value to authorizationHeaders argument in the MockClient constructor which is different to that of the httpOptions['headers'] argument. We should them assert that the value in the request headers is the one in httpOptions as it overrides the authorizationHeader argument.